incydr/cli/cmds/watchlists.py

```diff
@@ -0,0 +1,579 @@
+    watchlists = client.watchlists.v1.iter_all(user_id=user)
+    _output_results(watchlists, Watchlist, format_, columns)
+
+
+@watchlists.command(cls=IncydrCommand)
+@watchlist_arg
+@click.pass_context
+def show(ctx: Context, watchlist: str = None):
+    """
+    Show details for a watchlist.
+
+    WATCHLIST can be specified by watchlist type (ex: `DEPARTING_EMPLOYEE`) or ID.
+    `CUSTOM` watchlists must be specified by title or ID.
+
+    If using `rich`, outputs a summary of watchlist information and membership. This includes the following:
+
+    * included_users
+    * excluded_users
+    * included_departments
+    * included_directory_groups
+
+    Lists of users will be truncated to only display the first 25 members, use the `list-included-users` and `list-excluded-users`
+    commands respectively to see more details.
+
+    If not using `rich`, outputs watchlist information in JSON without additional membership summary information.
+    """
+    client = ctx.obj()
+    watchlist_response = client.watchlists.v1.get(watchlist)
+
+    if not client.settings.use_rich:
+        console.print(watchlist_response.json(), highlight=False)
+
+    def models_as_table(model, models, title=None):
+        headers = list(get_fields(model))
+        tbl = Table(*headers, title=title, show_lines=True)
+        for m in models:
+            values = []
+            for _name, value in iter_model_formatted(m, render="table"):
+                if isinstance(value, BaseModel):
+                    value = model_as_card(value)
+                elif not isinstance(value, (ConsoleRenderable, RichCast, str)):
+                    value = str(value)
+                values.append(value)
+            tbl.add_row(*values)
+        return tbl
+
+    included_users = client.watchlists.v1.list_included_users(watchlist).included_users
+    excluded_users = client.watchlists.v1.list_excluded_users(watchlist).excluded_users
+    departments = client.watchlists.v1.list_departments(watchlist).included_departments
+    dir_groups = client.watchlists.v1.list_directory_groups(
+        watchlist
+    ).included_directory_groups
+    t = Table(title=f"{watchlist_response.list_type} Watchlist")
+    t.add_column("Stats")
+
+    tables = []
+    if included_users:
+        tables.append(
+            models_as_table(WatchlistUser, included_users[:MAX_USER_DISPLAY_COUNT])
+        )
+        t.add_column("Included Users")
+    if excluded_users:
+        tables.append(
+            models_as_table(WatchlistUser, excluded_users[:MAX_USER_DISPLAY_COUNT])
+        )
+        t.add_column("Excluded Users")
+    if departments:
+        tables.append(models_as_table(IncludedDepartment, departments))
+        t.add_column("Included Departments")
+    if dir_groups:
+        tables.append(models_as_table(IncludedDirectoryGroup, dir_groups))
+        t.add_column("Included Directory Groups")
+
+    t.add_row(model_as_card(watchlist_response), *tables)
+
+    with console.pager():
```

Comment:
Prints out the watchlist summary as a table consisting of the `stats` column as well as columns for `included users, `excluded users`, `departments`, and `directory groups`.  Columns are omitted if the list is empty.

![Screen Shot 2022-11-22 at 2 05 30 PM](https://user-images.githubusercontent.com/81983309/203413279-3fe88ed6-b860-4a61-bfcb-d665ec8deca3.png)

Reply:
love it